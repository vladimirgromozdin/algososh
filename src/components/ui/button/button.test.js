import React from "react";
import renderer from 'react-test-renderer'

import {Button} from "./button";

it('should render button with text', () => {
    const tree = renderer
        .create(<Button text='Text'/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render button without text', () => {
    const tree = renderer
        .create(<Button />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render locked button', () => {
    const tree = renderer
        .create(<Button disabled={true}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render loading button', () => {
    const tree = renderer
        .create(<Button isLoader={true}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});