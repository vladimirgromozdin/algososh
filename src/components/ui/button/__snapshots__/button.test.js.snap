// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Text
  </p>
</button>
`;

exports[`should render button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`should render loading button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`should render locked button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
